test/runtests.jl

```diff
@@ -1,7 +1,32 @@
 using GPUArrays, Test
 using GPUArrays.TestSuite
 
+using Pkg
 
-@testset "Julia reference implementation:" begin
+@testset "JLArray" begin
     GPUArrays.test(JLArray)
 end
+
+function test_package(package)
+    mktempdir() do devdir
+        withenv("JULIA_PKG_DEVDIR" => devdir) do
+            # try to install from the same branch of GPUArrays
+            try
+                branch = chomp(read(`git -C $(@__DIR__) rev-parse --abbrev-ref HEAD`, String))
```

Comment:
`get rev-parse --abbrev-ref HEAD` will return `HEAD` in a detached state... as seen in `Installed CuArrays from HEAD branch`